packages/ui-color-picker/src/ColorPicker/index.tsx

```diff
@@ -0,0 +1,556 @@
+        hexCode,
+        isValidHex,
+        minContrast,
+        contrast
+      )
+    }
+
+    return [
+      ...invalidColorMessages,
+      ...isRequiredMessages,
+      ...generalMessages,
+      ...contrastMessages
+    ]
+  }
+  renderBeforeInput() {
+    const { styles } = this.props
+    return (
+      <div css={styles?.simpleColorContainer}>
+        {this.isSimple && <ColorIndicator color={`#${this.state.hexCode}`} />}
+        <div css={styles?.hashMarkContainer}>#</div>
+      </div>
+    )
+  }
+  renderAfterInput() {
+    const { checkContrast, styles } = this.props
+    const { hexCode } = this.state
+
+    if (checkContrast && isValid(hexCode)) {
+      const contrast = getContrast(
+        checkContrast.contrastAgainst || '#fff',
+        hexCode,
+        2
+      )
+      const minContrast = this.getminContrast(
+        checkContrast.contrastStrength ? checkContrast.contrastStrength : 'mid'
+      )
+
+      if (minContrast >= contrast) {
+        return (
+          <div css={styles?.errorIcons}>
+            {checkContrast.isStrict ? <IconTroubleLine /> : <IconWarningLine />}
+          </div>
+        )
+      }
+      return (
+        <div css={styles?.successIcon}>
+          <IconCheckDarkLine />
+        </div>
+      )
+    }
+    return null
+  }
+
+  handleOnChange(event: React.ChangeEvent<HTMLInputElement>, value: string) {
+    const { onChange } = this.props
+    if (
+      value.length > (this.props.withAlpha ? 8 : 6) ||
+      //TODO remove any
+      !acceptedCharactersForHEX.includes((event.nativeEvent as any).data)
+    ) {
+      return
+    }
+    if (typeof onChange === 'function') {
+      onChange(`${value}`)
+    } else {
+      this.setState({
+        showHelperErrorMessages: false,
+        hexCode: `${value}`,
+        mixedColor: `${value}`
+      })
+    }
+  }
+
+  //TODO remove any
+  handleOnPaste(event: any) {
+    const pasted = event.clipboardData.getData('Text')
+    let toPaste = pasted
+    if (pasted[0] && pasted[0] === '#') {
+      toPaste = pasted.slice(1)
+    }
+
+    if (
+      (this.props.withAlpha && toPaste.length > 8) ||
+      (!this.props.withAlpha && toPaste.length > 6)
+    ) {
+      return event.preventDefault()
+    }
+
+    const newHex = `${this.state.hexCode}${toPaste}`
+    if (isValid(newHex)) {
+      this.setState({
+        hexCode: newHex
+      })
+      return event.preventDefault()
+    }
+
+    event.preventDefault()
+  }
+
+  handleOnBlur() {
+    this.setState({ showHelperErrorMessages: true })
+  }
+
+  renderLabel() {
+    const { label, tooltip, styles } = this.props
+
+    return tooltip ? (
+      <div>
+        <span css={styles?.label}>{label}</span>
+        <span>
+          <Tooltip renderTip={tooltip}>
+            <IconInfoLine />
+          </Tooltip>
+        </span>
+      </div>
+    ) : (
+      label
+    )
+  }
+
+  stripAlphaIfNeeded = (hex: string) =>
+    hex.slice(-2) === 'FF' ? hex.slice(0, -2) : hex
```

Comment:
Just for future-proofing purpose, I would add the 8 char check, just in case it would be used on a 6char hex :D

```suggestion
  stripAlphaIfNeeded = (hex: string) =>
    hex.length === 8 && hex.slice(-2) === 'FF' ? hex.slice(0, -2) : hex
```